hydroflow_plus/src/builder.rs

```diff
@@ -19,8 +19,13 @@ use crate::location::{
 use crate::stream::{Async, Windowed};
 use crate::{HfCompiled, HfCycle, RuntimeContext, Stream};
 
+/// Tracks the leaves of the dataflow IR. This is referenced by
+/// `Stream` and `HfCycle` to build the IR. The inner option will
+/// be set to `None` when this builder is finalized.
+pub type FlowLeaves = Rc<RefCell<Option<Vec<HfPlusLeaf>>>>;
+
 pub struct FlowBuilder<'a, D: LocalDeploy<'a> + ?Sized> {
-    ir_leaves: Rc<RefCell<Vec<HfPlusLeaf>>>,
+    ir_leaves: FlowLeaves,
```

Comment:
I'm not sure exactly how this is being used but a new empty vec doesn't allocate, so the `Option` wrapping might be unnecessary

Reply:
Mostly for defensive coding, since empty Vec vs None represents two different states.